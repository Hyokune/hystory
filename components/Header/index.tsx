import Link from 'next/link'
import React from 'react'

class Header extends React.Component {
  render() {
    return (
      <header className="header">
        <div className="header-logo">
          <Link href="/">
            <a>Hystory</a>
          </Link>
        </div>
      </header>
    )
  }
}

export default Header
