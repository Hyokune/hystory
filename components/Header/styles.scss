@import '../../styles/base/index';

header {
  height: 2rem;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  padding: 0.6rem 1rem;
  background-color: $black-2;

  .header-logo {
    height: 100%;
    width: auto;
    display: flex;
    align-items: center;
    justify-content: center;

    img {
      height: auto;
      width: 3rem;
    }

    a {
      text-decoration: none;

      h1 {
        color: white;
        font-weight: 500;
        font-size: 1.4rem;
        letter-spacing: 0.5rem;
      }
    }
  }
}
