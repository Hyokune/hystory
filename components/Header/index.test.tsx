import { shallow } from 'enzyme'
import React from 'react'
import Header from './index'

describe('Header component', () => {
  test('render component without error', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.hasClass('header')).toBe(true)
  })
})
