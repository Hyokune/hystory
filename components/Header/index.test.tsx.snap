// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component render component without error 1`] = `
<header
  className="header"
>
  <div
    className="header-logo"
  >
    <Link
      href="/"
    >
      <a>
        Hystory
      </a>
    </Link>
  </div>
</header>
`;
